crates/btcio/src/reader/query.rs

```diff
@@ -105,6 +110,68 @@ async fn do_reader_task<R: Reader>(
     }
 }
 
+/// Checks for epoch and scan rule update. If so, rescan recent blocks to account for possibly
+/// missed relevant txs.
+async fn check_and_update_epoch_scan_rules_change<R: Reader>(
+    ctx: &ReaderContext<R>,
+    state: &mut ReaderState,
+) -> anyhow::Result<()> {
+    let latest_epoch = ctx.status_channel.epoch().unwrap_or(0);
+    // TODO: check if latest_epoch < current epoch. should panic if so?
+    let curr_epoch = match state.epoch() {
+        Epoch::Exact(e) => *e,
+        Epoch::AtTransition(_from, to) => *to,
+    };
+    let is_new_epoch = curr_epoch == latest_epoch;
+    let curr_filter_config = match state.filter_config() {
+        EpochFilterConfig::AtEpoch(c) => c.clone(),
+        EpochFilterConfig::AtTransition(_, c) => c.clone(),
+    };
+    if is_new_epoch {
+        // TODO: pass in chainstate to `derive_from`
+        let new_config = TxFilterConfig::derive_from(ctx.config.params.rollup())?;
+
+        // If filter rule has changed, first revert recent scanned blocks and rescan them with new
+        // rule
+        if new_config != curr_filter_config {
+            // Send L1 revert so that the recent txs can be appropriately re-filtered
```

Comment:
I couldn't think of better way to re-scan recent blocks without sending the revert event.